internal/provider/resource_consul_cluster.go

```diff
@@ -767,3 +832,37 @@ func resourceConsulClusterImport(ctx context.Context, d *schema.ResourceData, me
 
 	return []*schema.ResourceData{d}, nil
 }
+
+// buildIPAllowlist validates there are no duplicate CIDRs and returns a consul model.
+func buildIPAllowlist(cidrs []interface{}) ([]*consulmodels.HashicorpCloudConsul20210204CidrRange, error) {
```

Comment:
this function is only used to check for duplicates, right?

Reply:
Removed duplicate logic now, but it also converts the `ip_allowlist` from a terraform list (type `[]interface{}`) to a model that can be used by the consul client.

Used across a few operations so it's nice to have it